ci/precise_benchmarks.py

```diff
@@ -0,0 +1,150 @@
+import subprocess
+import os
+import sys
+import json
+from time import gmtime, strftime
+import platform
+import git
+
+if(len(sys.argv) != 2):
+    print("Usage: ./precise_benchmarks.py <ABS PATH TO PROCESS>")
+    exit(1);
+
+def run(cmd):
+    proc = subprocess.Popen(cmd,
+        stdout = subprocess.PIPE,
+        stderr = subprocess.STDOUT,
+    )
+    stdout, stderr = proc.communicate()
+    return proc.returncode, stdout, stderr
+def get_colon_index(string):
+    index = 0
+    found = False
+    for char in string:
+     if char == ':':
+         Found = True
+         return index+1
+     index = index + 1
+    if not found:
+     return -1
+
+def preprocess_brackets(string):
+    index = 0
+    string_has_bracket = False
+    for char in string:
+     if char == '(':
+         string_has_bracket = True
+         break
+     index = index + 1
+    if string_has_bracket:
+     return string[:index]
+    else:
+     return string
+
+def generic_processor(code, out):
+    generic_linesplit = []
+    for line in out.splitlines():
+     generic_linesplit.append(line)
+    generic_split = generic_linesplit[-15:] 
+    generic_results = []
+    for index, values in enumerate(generic_split):
+     token_index = get_colon_index(generic_split[index])
+     if token_index > 0:
+         token_value = generic_split[index][token_index:].strip()
+         preprocessed_token_value = preprocess_brackets(token_value)
+         generic_results.append(preprocessed_token_value)
+    return generic_results
+
+def create_json_from_dict(result_dict, file_name):
+    with open(file_name, "w") as json_dump_file:
+     json.dump(result_dict, json_dump_file, sort_keys=False, indent=4)
+
+def remove_comma(string):
+    returner = ""
+    for char in string:
+     if char != ',':
+         returner += str(char)
+     else:
+         continue
+    return returner
+
+def remove_percent(string):
+    returner = ""
+    for char in string:
+     if char != '%':
+         returner += str(char)
+     else:
+         continue
+    return returner
+
+def get_space_index(string):
+    index = 0
+    has_space = False
+    for char in string:
+     if char == ' ':
+         has_space = True 
+         index+1
+         break;
+     index = index + 1
+    if has_space:
+     return index
+    else:
+     return -1
+
+def percent_to_float(percent):
+    return float(percent/100)
```

Comment:
I think `/100` gives a float already, so there's no need to cast explicitly.
```suggestion
    return percent/100
```